runners/samza/src/main/java/org/apache/beam/runners/samza/translation/TranslationContext.java

```diff
@@ -249,6 +252,19 @@ public String getTransformId() {
     return idGenerator.getId(getTransformFullName());
   }
 
+  /** Given a set of user stateIds and parDo name, return a stateId to storeId map. */
+  public Map<String, String> getStateIdToStoreIdMap(Set<String> stateIds, String escapedParDoName) {
+    final Map<String, String> storeIds = new HashMap<>();
+    stateIds.forEach(
+        stateId ->
+            storeIds.put(
+                stateId,
+                multiParDoStateIds.contains(stateId)
+                    ? String.join("-", stateId, escapedParDoName)
```

Comment:
Let's create a helper method for this join, looks like it happens in a lot of place in the code.

Reply:
Done. Thanks!